### PR TITLE
enhancement(observability): add flag to send tracing spans to Datadog APM

### LINKED ISSUE #7641: Integrate tracing data with Datadog AMP
Vector has internal trace data but we don't terminate it anywhere. Terminate the soak traces into AMP.

### PR DESCRIPTION
**Pull request by lukesteensen**
Closes #7641 

This adds an option for sending tracing spans to a local Datadog agent (with APM enabled, which is not the default, see [docs](https://docs.datadoghq.com/agent/docker/apm/?tab=linux)). There is still some work to do to make the tracing data interesting, but I've added some new spans to the file source that would have come in very handy in previous investigations.

An important thing to note is that these spans respect the current log level, so if you want to see the new tracing spans from the file source you'll need to run with something like `LOG=info,file_source=trace`. This obviously makes the logs noisy as well, so we may want to do some thinking about how to choose levels. If the performance impact is low enough many of these could be bumped to debug or even info.

### CODE REVIEW

**Review by blt — APPROVED** [review]
Cool, so it looks like there's some check dings but I'm sold on this. Looking forward to getting it deployed in the soak infra.

**Review by spencergilbert — APPROVED** [review]

File: docs/reference/cli.cue [hunk 1]
```diff
@@ -103,6 +103,13 @@ cli: {
 			description: "Watch for changes in the configuration file, and reload accordingly"
 			env_var:     "VECTOR_WATCH_CONFIG"
 		}
+		"enable-datadog-tracing": {
+			description: """
+				  [experimental] Send internal tracing spans to a local APM-enabled
+				  Datadog agent with a granularity matching the current log level.
+				"""
+			env_var: "VECTOR_ENABLE_DATADOG_TRACING"
+		}
```
Comment (jszwedko) [inline docs/reference/cli.cue:112]: Do we anticipate that we could send these directly to DD in the future?

**Review by jszwedko — APPROVED** [review]
There are some clippy dings but otherwise 👍 I'm excited to see us expand tracing to more areas of Vector to make it easier to debug issues.

File: docs/reference/cli.cue (hunk 1 above)
Comment (lukesteensen) [reply docs/reference/cli.cue:112]: Yeah, once we have a direct integration for traces I imagine we would prefer that. My understanding is that it's a bit complex and therefore likely a little ways off.

**Pull request merged**